### PR TITLE
Add -dev-tls-san flag

### PR DESCRIPTION
**Pull request by tomhjp**
This is helpful when wanting to set up a dev server with TLS in Kubernetes and any other situations where the dev server may not be the same machine as the Vault client (e.g. in combination with some /etc/hosts entries)

### CODE REVIEW

**Review by jasonodonnell — APPROVED** [review]
Love this ❤️

File: command/server/tls_util.go [hunk 1]
```diff
@@ -80,6 +80,13 @@ func GenerateCert(caCertTemplate *x509.Certificate, caSigner crypto.Signer) (str
 	if !foundHostname {
 		template.DNSNames = append(template.DNSNames, hostname)
 	}
+	for _, san := range extraSans {
+		if ip := net.ParseIP(san); ip != nil {
```
[the remark this answers is not in the document]
Comment (tomhjp) [reply command/server/tls_util.go:84]: Ideally we could easily validate whether a `san` conforms to valid DNS name rules and then only accept values that are either a valid IP or a valid DNS name. However, I couldn't find an easy way to validate DNS names and figured the cost of allowing invalid DNS names (especially in the context of a dev flag) is very low anyway.

File: command/server.go [hunk 2]
```diff
@@ -971,7 +984,7 @@ func configureDevTLS(c *ServerCommand) (func(), *server.Config, string, error) {
 				return nil, nil, certDir, err
 			}
 		}
-		config, err = server.DevTLSConfig(devStorageType, certDir)
+		config, err = server.DevTLSConfig(devStorageType, certDir, c.flagDevTLSSans)
```
Comment (cipherboy) [inline command/server.go:987]: One comment if you've got a minute: do you want to add the listen address here as well? I think it'd help close https://github.com/hashicorp/vault/issues/18259, though we could always require the explicit SAN if we preferred... Just thinking the UX might be nice of the address+dev-tls (without dev-tls-san), but my 2c. :-)

**Review by cipherboy — APPROVED** [review]

File: command/server.go (hunk 2 above)
Comment (tomhjp) [reply command/server.go]: Nice idea! Added in 3deacb8c8e4323954b2c0e16613aea9d401f730a

File: command/server/tls_util.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (tomhjp) [reply command/server/tls_util.go:84]: Oh nice, thanks for sharing that. And yeah, also agreed on your conclusion 👍

**Comment by tomhjp** [comment]
Thanks both!

**Pull request merged**